Move ax mons palette to C and fix #34

DizzyEggg opened the pull request.

SethBarberee commented:
#34 is related to this but any luck with getting it in a `pal` like pokeemerald
https://github.com/pret/pokeemerald/blob/master/graphics/bag/bag_pyramid.pal

PAL file is significantly easier to work with and use in graphics programs tbh

DizzyEggg commented:
> #34 is related to this but any luck with getting it in a `pal` like pokeemerald https://github.com/pret/pokeemerald/blob/master/graphics/bag/bag_pyramid.pal
> 
> PAL file is significantly easier to work with and use in graphics programs tbh

I think #34 should be fixed now. I introduced a new palette format `.pmd_pal` which is basically `.pal`,  but with the 0x80 alpha flag. I also changed `gAxMonsPalette`, `banrpal` and the palette in `unk_data` to use the new format. I didn't remove the functionality of `.gbapal` for compatibility with pokeemerald and other gen3 games.
